### PR TITLE
Add blacklist for Request Cookies.

### PR DESCRIPTION
**Pull request by vitaliydavydiak**
Refactor code. Fix tests.

### CODE REVIEW

**Comment by StefH** [comment]
A question:

currently the blacklisted headers and cookies are just a string array.

Wouldn't it be better if an array/collection of [IStringMatcher](https://github.com/WireMock-Net/WireMock.Net/blob/master/src/WireMock.Net/Matchers/IStringMatcher.cs) would be allowed. When using a IStringMatcher, more complex matching like Wildcard or Regex are possible. What are your thoughts?

**Comment by vitaliydavydiak** [comment]
> A question:
> 
> currently the blacklisted headers and cookies are just a string array.
> 
> Wouldn't it be better if an array/collection of [IStringMatcher](https://github.com/WireMock-Net/WireMock.Net/blob/master/src/WireMock.Net/Matchers/IStringMatcher.cs) would be allowed. When using a IStringMatcher, more complex matching like Wildcard or Regex are possible. What are your thoughts?

Not sure if it brings that much value. Headers and Cookies are pretty unique and well-known so defining exact names collection is ok. Most likely you want to filter all that out.

File: src/WireMock.Net/Settings/IProxyAndRecordSettings.cs [hunk 1]
```diff
@@ -30,5 +30,10 @@ public interface IProxyAndRecordSettings
         /// Defines a list from headers which will excluded from the saved mappings.
         /// </summary>
         string[] BlackListedHeaders { get; set; }
+
+        /// <summary>
+        /// Defines a list of cookies which will excluded from the saved mappings.
+        /// </summary>
+        string[] BlackListedCookies { get; set; }
```
[the remark this answers is not in the document]
Comment (vitaliydavydiak) [reply src/WireMock.Net/Settings/IProxyAndRecordSettings.cs]: Done

**Comment by StefH** [comment]
A new NuGet will be created in some time...

**Pull request merged**